iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -418,6 +378,16 @@ void populateConvertSharedMemoryAllocOps(RewritePatternSet &patterns) {
   patterns.add<ConvertSharedMemAllocOp>(patterns.getContext());
 }
 
+void populateLowerHALInterfaceOp(RewritePatternSet &patterns) {
+  patterns.insert<HALInterfaceWorkgroupOpsConverter<
+                      IREE::HAL::InterfaceWorkgroupIDOp, gpu::BlockIdOp>,
+                  HALInterfaceWorkgroupOpsConverter<
+                      IREE::HAL::InterfaceWorkgroupSizeOp, gpu::BlockDimOp>,
```

Comment:
Do you really need this? I'm not sure you want to translate `InterfaceWorkgroupSizeOp` into `BlockDimOp`. `InterfaceWOrkgroupSizeOp` is [used as the tile size for workgroups](https://github.com/google/iree/blob/main/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp#L999). Although defined so, it's not used as the number of threads in a block (unless each thread just handles one scalar) right now. It's quite confusing, and this is one thing we should resolve..

Reply:
You're right, I don't need it and it is wrong. This get replaced much before. Removed it.